### ISSUE TITLE
🐛 BUG: svelte onMount isn't called with client:only

### ISSUE THREAD

**Issue by locriacyber** [snowpackjs/astro#1606]
### What package manager are you using?

pnpm

### What operating system are you using?

Linux

### Describe the Bug

index.astro:
```
---
import Counter from '../components/Counter.svelte';
---
<html lang="en">
<head>
    <meta charset="UTF-8">
</head>
<body>
    <main>
        <Counter client:only />
    </main>
</body>
</html>
```

Counter.svelte:
```
<script>
  import { onMount } from 'svelte'

  let count = 1

  onMount(() => {
    count = 2
  })
</script>
```

### Steps to Reproduce

1. `pnpm init astro` using starter template (first option), add svelte renderer
2. replace `index.astro` with above code
3. add `src/components/Counter.svelte`
4. Error! Page shows 1, not 2.

**Comment by matthewp** [comment]
Marking this to be fixed in compiler-v2, which is in the process of getting support for `client:only`.

**Comment by FredKSchott** [comment]
Thanks for the report! We believe that this has been fixed in astro v0.21, going out today. Let us know if you can still reproduce the issue after updating to v0.21 and we'll be happy to take a look.
